src/remote/watcher.coffee

```diff
@@ -77,8 +79,7 @@ class RemoteWatcher
                         @whenReady callback
             else
                 @changes = @couch.client.changes
-                    filter: (doc) ->
-                        doc.docType?.toLowerCase() in ['file', 'folder']
+                    filter: @filterName
```

Comment:
I don't understand what you find more complicated. It's just how the PouchDB API works. You can see an example on https://pouchdb.com/api.html#filtered-changes (example 5).